### ISSUE TITLE
Missing var reference in Show rule Documentation's anonymous function

### ISSUE THREAD

**Issue by rtbs-dev** [typst/typst#2565]
https://github.com/typst/typst/blame/b716700b61b229e2b703b892d3e047d84633d982/docs/reference/styling.md#L93

Missing an `#` for `#it` in the body of the anonymous function.

**Comment by PgBiel** [comment]
The code is compiling, so there are no issues here.

Note that `#` doesn't mean "fetch the value of a variable", but rather "enter code mode" (where any identifier is read as a variable). So, by doing `#emph`, we're already in code mode inside its arguments, so we can already write `#emph(it.body)` without any additional `#` (note that writing `#emph(#it.body))` will produce an error precisely because you are already in code mode).

See also https://typst.app/docs/reference/scripting/

**Comment by rtbs-dev** [comment]
@PgBiel Ah, apologies, I just realized mine wasn't using the variable because I had called `#emph[it.body]` rather than `#emph(it.body)`, which means it was being treated as content already. 

ngl, still a bit confused when the functions are going to use `()` or `[]`, mostly because it seems the line between element constructor and content function is getting blurry in my head, I guess? Still that's a "me" problem, thanks for the quick response. I was coming here to close once I noticed and you beat me to it! ^_^
